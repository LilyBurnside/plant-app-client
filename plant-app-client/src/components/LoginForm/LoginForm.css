form {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

div.user_name, div.password {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

div > input {
  border: none;
  border-radius: 5px;
  box-shadow: 2px 2px #5A7F35;
  padding-top: 5px;
  padding-left: 5px;
  margin: 0 50px 10px 50px;
}

div > label {
  padding: 10px;
}

button {
  color: #494949 !important;
  text-decoration: none;
  background: white;
  padding: 10px 20px;
  border: 4px solid #494949 !important;
  margin-top: 50px;
  border-radius: 15px;
}

button:hover {
  background-color: #D58BFA;
  color: white !important;
  transition: .25s;
}

.title h1, .title p, .title a {
  background-color: #DBFFB7;
}

.title p {
  padding: 0px 40px;
}