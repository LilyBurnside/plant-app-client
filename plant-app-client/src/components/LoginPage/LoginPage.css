.LoginPage {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: #DBFFB7;
  border-radius: 15px;
  padding: 50px 10px 75px 10px;
  margin-top: 50px;
}