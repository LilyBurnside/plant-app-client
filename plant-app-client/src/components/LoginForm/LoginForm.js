import React from 'react'
import TokenService from '../../services/token-service'
import AuthApiService from '../../services/auth-api-service'
import QuizContext from '../../contexts/QuizContext'
import './LoginForm.css'

export default class LoginForm extends React.Component {
  static defaultProps = {
    onLoginSuccess: () => {}
  }

  state = { error: null }

  static contextType = QuizContext


  handleSubmitJwtAuth = ev => {
    ev.preventDefault()
    this.setState({ error: null })
    const { user_name, password } = ev.target
    AuthApiService.postLogin({
      user_name: user_name.value,
      password: password.value,
    })
      .then(res => {
        user_name.value = ''
        password.value = ''
        TokenService.saveAuthToken(res.token)
        this.props.onLoginSuccess()
        this.context.setUserId(res.payload.user_id)
      })
      .catch(this.context.setError)
    
  }

  render() {
    const { error } = this.state
    return (
      <form
        className='loginForm'
        onSubmit={this.handleSubmitJwtAuth}
      >
        <div role='alert'>
          {error && <p className='red'>{error}</p>}
        </div>
        <div className='user_name'>
          <label htmlFor='loginForm-user_name'>
            User name
          </label>
          <input
            required
            className='input-loginForm'
            name='user_name'
            id='loginForm-user_name'>
          </input>
        </div>
        <div className='password'>
          <label htmlFor='loginForm-password'>
            Password
          </label>
          <input
            required
            className='input-loginForm'
            name='password'
            type='password'
            id='loginForm-password'>
          </input>
        </div>
        <button className='button-loginForm' type='submit'>
          Login
        </button>
      </form>
    )
  }
}