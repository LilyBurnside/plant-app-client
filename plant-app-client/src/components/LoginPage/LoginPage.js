import React, { Component } from 'react'
import LoginForm from '../LoginForm/LoginForm'
import './LoginPage.css'

export default class LoginPage extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  }

  handleLoginSuccess = () => {
    const { location, history } = this.props
    const destination = (location.state || {}).from || '/'
    history.push(destination)
    window.location.reload()
  }

  render() {
    return (
      <section className='LoginPage'>
        <h2>Login</h2>
        <LoginForm
          onLoginSuccess={this.handleLoginSuccess}
        />
      </section>
    )
  }
}