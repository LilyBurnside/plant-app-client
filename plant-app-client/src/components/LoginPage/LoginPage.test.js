import React from 'react';
import ReactDom from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoginPage from './LoginPage';
import LoginForm from '../LoginForm/LoginForm'

describe('Login Page Component', () => {
  
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(
      <LoginPage />,
      div
    );
    ReactDom.unmountComponentAtNode(div);
  });

  it('renders Login Form', () => {
    const wrapper = shallow(
      <LoginPage>
        <LoginForm/>
      </LoginPage>
    );
    
    expect(toJson(wrapper)).toMatchSnapshot();
  })

})