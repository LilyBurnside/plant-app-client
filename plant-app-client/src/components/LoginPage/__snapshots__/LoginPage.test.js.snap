// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page Component renders Login Form 1`] = `
<section
  className="LoginPage"
>
  <h2>
    Login
  </h2>
  <p>
    Log in with username: guest and password: password
  </p>
  <LoginForm
    onLoginSuccess={[Function]}
  />
</section>
`;
